src/stdLambda.cls

```diff
@@ -747,11 +764,143 @@ Private Sub parseFlowPriority1()
         
             Call optConsume("end")
         End If
+    Else
+        Call parseFlowPriority2
+    End If
+End Sub
+
+'Parse loops (do ... until|while ... loop)
+Private Sub parseFlowPriority2()
+    If optConsume("do") Then
+        Dim RepeatLoopIndex As Integer: RepeatLoopIndex = This.iOperationIndex
+        Dim RepeatLoopJumpIndex As Integer
+        Dim ExitLoopJumpIndex As Integer: ExitLoopJumpIndex = -1 ' -1 to ensure a value, which means "no exit"
+        If optConsume("until") Then
+            Call parseExpression                                     ' Break Condition
+            RepeatLoopJumpIndex = addOperation(oJump, ifTrue, , -1)  ' Skip loop
+        ElseIf optConsume("while") Then
+            Call parseExpression                                     ' Break Condition
+            RepeatLoopJumpIndex = addOperation(oJump, ifFalse, , -1) ' Skip loop
+        Else
+            ' Infinite loop as long as no exit do is used
+        End If
+        Call parseBlock("exit", "loop")
+        If optConsume("exit") Then
+            ExitLoopJumpIndex = addOperation(oJump, , , -1) ' Skip loop
+            Call parseBlock("loop")
+        End If
+        Call addOperation(oJump, , RepeatLoopIndex, -1)
+        Call consume("loop")
+        If ExitLoopJumpIndex <> -1 Then This.operations(ExitLoopJumpIndex).value = This.iOperationIndex
+        This.operations(RepeatLoopJumpIndex).value = This.iOperationIndex
+    Else
+        Call parseFlowPriority3
+    End If
+End Sub
+
+'Parse loops (for ... to ... step ... next)
+Private Sub parseFlowPriority3()
+    Dim varName As String, Operator As String, OperatorValue As String, Increment As String
+    If optConsume("for") Then
+        Call ForLoopLet()

```

Comment:
Not too big a fan of for loop being spread over many function calls. The functions that were created aren't really re-usable for anything other than the for-loop in the end, so it feels unnecessary in my head?